runtime/ms-rest-azure/lib/filters/rpRegistrationFilter.js

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information. 
+
+'use strict';
+
+var request = require('request');
+var utils = require('../utils');
+
+exports.retryTimeout = 30;
+
+/**
+ * Creates a filter that verifies whether the request failed due to RP not being registered.
+ * It tries to register the RP and after successful registration it retries the original request.
+ * @param {number} retryTimeoutInSec The retry timeout in seconds. Default is 30 seconds.
+ */
+exports.create = function create(retryTimeoutInSec) {
+
+  if (retryTimeoutInSec !== null && retryTimeoutInSec !== undefined) {
+    exports.retryTimeout = retryTimeoutInSec;
+  }
+  return function handle (options, next, callback) {
+    return next(options, function (err, response, body) {
+      if (err) {
+        return callback(err);
+      }
+      let rpName, urlPrefix;
+      if (response.statusCode === 409) {

```

Comment:
consider: extract `409` to a constant with an appropriate name like `RP Not registered`. That way, code is self documenting and cleaner.